### ISSUE TITLE
Rockets stack up

### ISSUE THREAD

**Issue by teccheck** [yuwui/Uchu#289]
If you build a new custom rocket it stacks on top of the old/current rocket, no matter what parts you use. I watched a [video](https://www.youtube.com/watch?v=FYqjZBnuBIg&list=PL9EFB8500E625ADC7&index=20) of the original gameplay and this is not the intended behavior. This is probably related to #63 and #157.

I don't know the code responsible for that system, but I suspect that there needs to be a check for the containing parts when putting a rocket into the inventory, so that a new instance will be created.

**Comment by teccheck** [comment]
The problem is not that there is a missing check, it's that `stackSize` has the value `2147483647` when its checked in `AddLotAsync` in `InventoryManagerComponent.cs:`. So there is probably an error while getting the stack size, because it has the right value in the database

**Comment by teccheck** [comment]
This code [snippet](https://github.com/UchuServer/Uchu/blob/8b579acaebcb13428b19482ce9a4c8133c34c7da/Uchu.World/Objects/Components/Player/InventoryManagerComponent.cs#L361) is the source of this problem:
```c#
// Some items have no stack size, like bricks
var stackSize = itemComponent.StackSize ?? 1;
if (stackSize == default) 
    stackSize = int.MaxValue;
```
If `stackSize` is 0 then it will be set to `int.MaxValue`. That is all good for bricks, as the comment explains but rockets do have an **actual** stackSize of 0.

I would rather check if the item is of type `Brick`, then set stackSize to `int.MaxValue`. You could also check for type `Vehicle` and set the stackSize to 0.

**Comment by enteryournamehere** [comment]
I believe the issue is because Uchu doesn't have subkeys implemented, which are meant to distinguish modular build items (rockets, cars) from each other. It is used in the GameMessages for moving items in inventories (for example see [AddItemToInventoryClientSync](https://github.com/UchuServer/Uchu/blob/caa7be2799eacba313f0c30835e50adb3d5ddf05/Uchu.World/Packets/GameMessages/Server/AddItemToInventoryClientSyncMessage.cs#L20), it's also in the GMs for moving items between inventories, deleting an item from an inventory and for notifying the client about a reward that has been mailed to them).

**Comment by teccheck** [comment]
I'm not quite sure... For testing I added an if statement to directly create a new stack when a rocket is added to the inventor and it worked.

Btw: Do you know if there are any item types that have a stackSize of 0 when you can have an unlimited amount of them (like bricks)?

**Comment by enteryournamehere** [comment]
Subkeys were definitely used for modular builds (and pets), but I'm not sure which things can or can't work without them.

> Btw: Do you know if there are any item types that have a stackSize of 0 when you can have an unlimited amount of them (like bricks)?

Yes, this is the case for all bricks, see for example: https://lu.lcdruniverse.org/explorer/objects/29/11.

**Comment by teccheck** [comment]
I'm quite new to this. How do Subkeys work, what are they for?

**Comment by enteryournamehere** [comment]
AFAIK their purpose is to distinguish between different items that have the same LOT, so that's each rocket / car you build and each pet of the same type you tame. I think it's meant to be randomly generated by the server when you get one of those items. (A while ago I wrote some initial code for this, but it's been a while and I'm not sure how much I tested.)

**Comment by enteryournamehere** [comment]
Fixed in #290
